smart-alluxio/src/main/java/org/smartdata/alluxio/metric/fetcher/AlluxioNamespaceFetcher.java

```diff
@@ -0,0 +1,298 @@
+          }
+          this.currentBatch = new FileStatusInternalBatch(DEFAULT_BATCH_SIZE);
+        }
+
+        if (this.batches.isEmpty()) {
+          this.isFinished = true;
+          long curr = System.currentTimeMillis();
+          LOG.info(String.format(
+              "Finished fetch Namespace! %d secs used, numDirs = %d, numFiles = %d",
+              (curr - startTime) / 1000,
+              numDirectoriesFetched, numFilesFetched));
+        }
+        return;
+      }
+
+      try {
+        URIStatus status = fs.getStatus(new AlluxioURI(parent));
+        if (status != null && status.isFolder()) {
+          List<URIStatus> children = fs.listStatus(new AlluxioURI(parent));
+          FileStatusInternal internal = convertToFileStatusInternal(status, children.size());
+          internal.setPath(parent);
+          this.addFileStatus(internal);
+          numDirectoriesFetched++;
+
+          for (URIStatus child : children) {
+
+            if (child.isFolder()) {
+              this.deque.add(child.getPath());
+            } else {
+              this.addFileStatus(convertToFileStatusInternal(child, 0));
+              numFilesFetched++;
+            }
+          }
+        }
+      } catch (IOException | InterruptedException | AlluxioException e) {
+        LOG.error("Totally, numDirectoriesFetched = " + numDirectoriesFetched
+            + ", numFilesFetched = " + numFilesFetched
+            + ". Parent = " + parent, e);
+      }
+    }
+
+    public boolean finished() {
+      return this.isFinished;
+    }
+
+    public FileStatusInternalBatch pollBatch() {
+      return this.batches.poll();
+    }
+
+    public void addFileStatus(FileStatusInternal status) throws InterruptedException {
+      this.currentBatch.add(status);
+      if (this.currentBatch.isFull()) {
+        this.batches.put(currentBatch);
+        this.currentBatch = new FileStatusInternalBatch(DEFAULT_BATCH_SIZE);
+      }
+    }
+
+    private FileStatusInternal convertToFileStatusInternal(URIStatus status,
+        int numOfChildren, String parent) {
+      FileStatusInternal internal = new FileStatusInternal(
+          status.getLength(), 
+          status.isFolder(),
+          1,
+          status.getBlockSizeBytes(),
+          status.getLastModificationTimeMs(),
+          status.getCreationTimeMs(),
+          new FsPermission((short) status.getMode()) , 
+          status.getOwner(),
+          status.getGroup(),
+          null,
+          status.getPath().substring(1, status.getPath().length()).getBytes(),
+          parent,
+          status.getFileId(),
+          numOfChildren,
+          null,
+          (byte) 0);
+      return internal;
+    }
+    
+    private FileStatusInternal convertToFileStatusInternal(URIStatus status,
+        int numOfChildren) {
+      return convertToFileStatusInternal(status, numOfChildren,"");
+    }
+ 
+
+  }
+
+  private static class FileStatusConsumer implements Runnable {
```

Comment:
Looks like this class can also be a util class shared by both Hdfs and Alluxio module